src/Features/Blockcore.Features.ColdStaking/ColdStakingManager.cs

```diff
@@ -632,7 +632,7 @@ public override bool ProcessTransaction(Transaction transaction, int? blockHeigh
                 if (script.IsUnspendable)
                 {
                     var data = TxNullDataTemplate.Instance.ExtractScriptPubKeyParameters(script);
-                    if (data.Length == 1)
+                    if (data != null && data.Length == 1)
```

Comment:
ah interesting, in what cases you happen to come across that it can be null?

Reply:
It was on testnet so might have been malformed transactions or something